src/containers/lineup/LineupProvider.tsx

```diff
@@ -562,7 +562,7 @@ class LineupProvider extends PureComponent<CombinedProps, LineupProviderState> {
     if (
       isMetadataLoading &&
       lineup.hasMore &&
-      tiles.length < (count !== undefined ? count : MAX_TILES_COUNT) &&
+      tiles.length < (count !== undefined ? count : lineupCount) &&
```

Comment:
This seems like the right change but I'm hesitant since this file is really confusing and I'm not sure what side-effects I might be causing cause of this. This change was put in to prevent infinite loading of tracks since count = profile.track_count doesn't account for hidden tracks so it keeps trying to get more even though no more tracks are there.

Reply:
We *really* need to rewrite lineup. @jowlee can you look into how safe this change may be?

I'm ok with this, and we shouldn't stop you from merging, but we should test pretty hard-core on staging.audius.co